mavros_extras/src/plugins/distance_sensor.cpp

```diff
@@ -198,45 +198,48 @@ class DistanceSensorPlugin : public plugin::PluginBase {
 		auto it = sensor_map.find(dist_sen.id);
 		if (it == sensor_map.end()) {
 			ROS_ERROR_NAMED("distance_sensor",
-					"DS: no mapping for sensor id: %d, type: %d, orientation: %d",
-					dist_sen.id, dist_sen.type, dist_sen.orientation);
+						"DS: no mapping for sensor id: %d, type: %d, orientation: %d",
+						dist_sen.id, dist_sen.type, dist_sen.orientation);
 			return;
 		}
 
 		auto sensor = it->second;
 		if (sensor->is_subscriber) {
 			ROS_ERROR_NAMED("distance_sensor",
-					"DS: %s (id %d) is subscriber, but i got sensor data for that id from FCU",
-					sensor->topic_name.c_str(), sensor->sensor_id);
+						"DS: %s (id %d) is subscriber, but i got sensor data for that id from FCU",
+						sensor->topic_name.c_str(), sensor->sensor_id);
 			return;
 		}
 
 		if (sensor->orientation >= 0 && dist_sen.orientation != sensor->orientation) {
 			ROS_ERROR_NAMED("distance_sensor",
-					"DS: %s: received sensor data has different orientation (%s) than in config (%s)!",
-					sensor->topic_name.c_str(),
-					utils::to_string_enum<MAV_SENSOR_ORIENTATION>(dist_sen.orientation).c_str(),
-					utils::to_string_enum<MAV_SENSOR_ORIENTATION>(sensor->orientation).c_str());
+						"DS: %s: received sensor data has different orientation (%s) than in config (%s)!",
+						sensor->topic_name.c_str(),
+						utils::to_string_enum<MAV_SENSOR_ORIENTATION>(dist_sen.orientation).c_str(),
+						utils::to_string_enum<MAV_SENSOR_ORIENTATION>(sensor->orientation).c_str());
 		}
 
 		auto range = boost::make_shared<sensor_msgs::Range>();
 
 		range->header = m_uas->synchronized_header(sensor->frame_id, dist_sen.time_boot_ms);
 
-		range->min_range = dist_sen.min_distance * 1E-2; // in meters
+		range->min_range = dist_sen.min_distance * 1E-2;// in meters
 		range->max_range = dist_sen.max_distance * 1E-2;
 		range->field_of_view = sensor->field_of_view;
 
-		if (dist_sen.type == enum_value(MAV_DISTANCE_SENSOR::LASER)) {
+		switch (dist_sen.type) {
+		case enum_value(MAV_DISTANCE_SENSOR::LASER):
+		case enum_value(MAV_DISTANCE_SENSOR::RADAR):
+		case enum_value(MAV_DISTANCE_SENSOR::UNKNOWN):
```

Comment:
the most common type on copter are 1-D Lidar that are laser type and IR on ROS. So that is why I chose this. Another posibillity could be to add another parameter on the config file. What do you think?